File: 08-Testing-CSharp/Solution/Testing/MyClass.cs
namespace Testing;

public sealed class MyClass(ILogger logger)
{
    public void DoLoop()
    {
        logger.Log("Before loop");

        for (var i = 1; i <=3; i++)
        {
            logger.Log($"In loop: {i}");
        }

        logger.Log("After loop");
    }
}

File: 08-Testing-Java/Solution/src/testing/MyClass.java
package testing;

public class MyClass {
    private final ILogger logger;

    public MyClass(ILogger logger) {
        this.logger = logger;
    }

    public void doLoop() {
        logger.log("Before loop");

        for (int i = 1; i <= 3; i++) {
            logger.log("In loop: " + i);
        }

        logger.log("After loop");
    }
}
